File: check_brackets.cpp
#include <iostream>
#include <stack>
#include <string>
#include <set>

using namespace std;

bool check_brackets(string str) {

    stack <char> mystack;
    set <char> opening_brackets = {'(', '{', '['};

    for (int i = 0; i < str.length(); i++) {

        auto it = opening_brackets.find(str[i]);

        if (it != opening_brackets.end()) {
            mystack.push(*it);
        } else if (mystack.empty()) {
            return false;
        } else {
            char top = mystack.top();
            mystack.pop();

            if ((top == '[' and str[i] != ']') ||
                (top == '(' and str[i] != ')') ||
                (top == '{' and str[i] != '}')) {
                return false;
            }
        }
    }
    return mystack.empty();
}


int main(int argc, char* argv[]) {
    if (check_brackets(argv[1])) {
        cout << "True" << endl;
    }
    else {
        cout << "False" << endl;
    }
}

File: check_brackets.py
import sys

def check_brackets(str):
    stack = []
    brackets = {'{', '(', '['}
    for char in str:
        if char in brackets:
            stack.append(char)
        elif not stack:
            return False
        else:
            top = stack.pop()
            if (top is '[' and char is not ']') or \
                    (top is '(' and char is not ')' ) or \
                    (top is '{' and char is not '}'):
                return False
    return not stack


if __name__ == '__main__':
    print(check_brackets(sys.argv[1]))